web/handlers.go

```diff
@@ -118,21 +119,20 @@ func generateDevCSP(c Context) string {
 			// Honour only supported keys
 			switch devFlagKey {
 			case "unsafe-eval", "unsafe-inline":
-				devCSPMap[devFlagKey] = true
+				devCSP = append(devCSP, "'"+devFlagKey+"'")
 			default:
 				c.Logger.Warn("Unrecognized developer flag", mlog.String("developer_flag", devFlagKVStr))
 			}
 		}
 	}
-	var devCSP string
-	supportedCSPFlags := []string{"unsafe-eval", "unsafe-inline"}
-	for _, devCSPFlag := range supportedCSPFlags {
-		if devCSPMap[devCSPFlag] {
-			devCSP += fmt.Sprintf(" '%s'", devCSPFlag)
-		}
+
+	// Add flags for Webpack dev servers used by other products during development
+	if model.BuildNumber == "dev" {
+		// Focalboard runs on http://localhost:9006
+		devCSP = append(devCSP, "http://localhost:9006")
```

Comment:
Let's not hardcode this. Ideally, I'd want some way for the boards module to expose which port their webserver is going to run via the server side MPA. Or for now, let's just use an env var.

Reply:
I could see having that done through the MPA, but I'd rather not use an env var since I'd like something that just works out of the box.

Would it be okay leaving that for a followup since this is only added for dev builds?

Reply:
I think we can come up with a solution without env vars. Can you point me to where in the focalboard codebase is the port 9006 set? My grep is failing me.

Reply:
I forgot to add the related PRs to this ticket. It's referenced in the new webpack config for both [the web app](https://github.com/mattermost/mattermost-webapp/pull/11001/files#diff-1fb26bc12ac780c7ad7325730ed09fc4c2c3d757c276c3dacc44bfe20faf166fR421) and [focalboard](https://github.com/mattermost/focalboard/pull/3748/files#diff-1c5589d2b7076ec8b90835ff95b065e6f57d7330758e104e1f853e079368c7c2R194).

Maybe this is a point in favour of switching the web app over to only supporting running through the Webpack Dev Server since then we wouldn't need any special handling for the dev case here...

Reply:
Okay perfect, so let's just externalize the port for now. We can just let the Makefile target [here](https://github.com/mattermost/focalboard/pull/3748/files#diff-d9ed00560219f02467267ef85ff915bf43d30687f7dcef88893df60bb19ee0c9R179) pass the port from there. Hardcoding in server is okay for now. But later we'd just set it as a focalboard config and get that config value on the server.

Reply:
Maybe we should chat more about what this does. This URL isn't something really configured on the server since it's only used to serve up files during development. It doesn't really make sense to have the MM server involved in that since it isn't going to be responsible for doing that for products any more. The only reason it has to know about it here is because it serves files for the web app during development which might end up changing anyway to simplify the web app developer flow.

Also, even if it were configured in the server, that's not really something we have access to from Webpack before it's built the web portion of Boards. We could technically have Webpack contact the server to get the config, but that seems unnecessary.

Reply:
Sure, we can chat. I am not asking it to be added in the server. I just suggested for the port to be externalized in the Makefile of focalboard [here](https://github.com/mattermost/focalboard/pull/3748/files#diff-d9ed00560219f02467267ef85ff915bf43d30687f7dcef88893df60bb19ee0c9R179).